sympy/physics/mechanics/particle.py

```diff
@@ -223,3 +223,38 @@ def potential_energy(self, scalar):
         """
 
         self._pe = sympify(scalar)
+
+    def total_energy(self, frame):
+        """The total energy of the Particle.
+
+        The total energy, E, of a particle, P, is given by
+
+        'E = K.E + P.E.'
+
+        where P.E. is the potential energy of the particle P, and
+        K.E. is kinetic energy of the particle P, in the supplied ReferenceFrame.
+
+        Parameters
+        ==========
+
+        frame : ReferenceFrame
+            The Particle's velocity is typically defined with respect to
+            an inertial frame but any relevant frame in which the velocity is
+            known can be supplied.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.mechanics import Particle, Point, ReferenceFrame
+        >>> from sympy import symbols
+        >>> m, v, r, g, h = symbols('m v r g h')
+        >>> N = ReferenceFrame('N')
+        >>> O = Point('O')
+        >>> P = Particle('P', O, m)
+        >>> P.point.set_vel(N, v * N.y)
+        >>> P.potential_energy = m * g * h
+        >>> P.total_energy(N)
+        g*h*m + m*v**2/2
+
+        """
+        return self.kinetic_energy(frame) + self.potential_energy
```

Comment:
Should raise an error if potential_energy is not defined.

Reply:
At present, potential_energy for a body for which potential_energy is not explicitly defined is taken as zero. I think this makes sense. What do you think about it?

Reply:
It is a bit misleading. If I type "total_energy" I'm expecting to get KE + PE, but if PE is zero by default I'm not getting any PE contribution. I feel like it would be best to inform the user that they aren't getting what they ask for. Maybe a warning message that tells the user that PE is zero and not set is sufficient.